### ISSUE TITLE
helm CLI doesnt install/fetch specific version when run with  "--version" option

### ISSUE THREAD

**Issue by hayderimran7** [helm/helm#4008]
Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.0", GitCommit:"fc32d2f3698e36b93322a3465f63a14e9f0eaead", GitTreeState:"clean", BuildDate:"2018-03-26T16:55:54Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.0", GitCommit:"fc32d2f3698e36b93322a3465f63a14e9f0eaead", GitTreeState:"clean", BuildDate:"2018-03-26T16:44:10Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
**Minikube** 

**Description**:
When running `helm fetch` or `helm install` to install/fetch a specific versioned chart using `--version=` option, helm CLI doesnt respect the passed version to `--version` and instead always installs latest one.  
In other words, `--version=` is not working as expected.  
This was tested where charts were uploaded to chartmuseum. 

**Steps To Reproduce**

1. Start a local chartmuseum server:
```
docker run \
       -d \
       --name chartmuseum \
        -p 8088:8080  \
        -e STORAGE="local" \
        -e STORAGE_LOCAL_ROOTDIR="./chartstorage" \
      chartmuseum/chartmuseum
```
2. Create a simple helm chart `my-app`
```
helm create my-app
```
3. Package the chart with one version:
```
helm package my-app/ --version=1.0.0-dev1+1
```
4. Package same chart again with different version:
```
helm package my-app/ --version=1.0.0-dev1+2
```
5. Upload both charts to local chartmuseum 
```
curl --data-binary "@my-app-1.0.0-dev1+2.tgz" http://localhost:8088/api/charts
curl --data-binary "@my-app-1.0.0-dev1+1.tgz" http://localhost:8088/api/charts
```
6. Add the chartmuseum as helm repo:
```
helm repo add cmtest http://localhost:8088/
```
7. Refresh the indexes:
```
helm repo update
```
8. Search for the chart uploaded `my-app`
```
 helm search cmtest/my-app
NAME         	CHART VERSION	APP VERSION	DESCRIPTION
cmtest/my-app	1.0.0-dev1+1 	1.0        	A Helm chart for Kubernetes
```
9. Now try to get chart of different version as shown in `helm search cmtest/my-app` i.e. `--version=1.0.0-dev1+2`
```
mkdir testfetch
cd testfetch
helm fetch cmtest/my-app --version=1.0.0-dev1+2
ls
my-app-1.0.0-dev1+1.tgz
```
🔴  the fetched package is `my-app-1.0.0-dev1+1` but specified `--version=1.0.0-dev1+2`. 

Its the same behavior when doing an install also 😞

**Comment by adamreese** [comment]
run with flag `--devel` to fetch unreleased versions

**Comment by hayderimran7** [comment]
@adamreese  still no luck .
```
$ rm -rf *
$ helm fetch --devel cmtest/my-app --version=1.0.0-dev1+2
$ ls
my-app-1.0.0-dev1+1.tgz
```

**Comment by natcho** [comment]
We are facing exactly the same problem where helm is handling versions incorrectly when the difference in versions is in the build metadata (even though the build metadata is sortable in the expected order). This applies to helm search, helm fetch, helm install.

**Comment by ivan-vinitskyy** [comment]
This seems quite serious.  If I want to install version 1.2.3, I can't do it if someone uploaded a chart with version 1.2.3+1.  It almost means I can never trust what I supply as --version parameter because ultimate decision sits with whatever the repository holds.

**Comment by bacongobbler** [comment]
I believe this has been fixed in https://github.com/helm/helm/pull/4850 which is available as of Helm 2.13.0. Closing for now, but please re-open if I'm mistaken. Thanks!

**Comment by Keith-Nowosielski-SJSU** [comment]
I'm having this issue happen with pull/fetch. 
Using search and specifying the version it can locate it, however using pull with the same version I get this error: 

helm pull repo rancher-stable/rancher --version 2.7.3
Error: non-absolute URLs should be in form of repo_name/path_to_chart, got: repo


version.BuildInfo{Version:"v3.11.3", GitCommit:"323249351482b3bbfc9f5004f65d400aa70f9ae7", GitTreeState:"clean", GoVersion:"go1.20.3"}
